mne_nirs/tests/test_examples.py

```
@@ -26,7 +27,10 @@ def examples_path():
 
 
 def run_script_and_check(test_file_path):
-    with open(test_file_path) as fid:
+    with open(test_file_path) as fid, warnings.catch_warnings():
+        # Ignore deprecation warning caused by
```

I added this workaround in order to attempt to fix these kind of [failures](https://dev.azure.com/mne-tools/mne-nirs/_build/results?buildId=24870&view=logs&jobId=8e630eb1-cb11-5b3c-6e81-b13952459f99&j=8e630eb1-cb11-5b3c-6e81-b13952459f99&t=923c8314-bb6c-5f49-b147-9d7b6ecb81f8) which are caused by a warning that originates from https://github.com/mne-tools/mne-python/blob/cecbf0fb1dc919254bb6095db491b2d7c5f68003/mne/viz/backends/_utils.py#L164
Do you have a better suggestion on how to mitigate this? Maybe we could submit a patch to upstream?

---

Yes a fix in MNE-Python would be appreciated!

---

What should a fix look like for this? I guess we can't remove the code because MNE-Python still needs to support previous versions of Qt? And it feels dangerous to catch and ignore the errors in MNE-Python. Any suggestions @larsoner ?

---

In MNE we should check the Qt version and set this attribute conditionally. Or do whatever the Qt deprecation says we should do for new enough versions (set a different attribute, no need to set anything, etc. -- I haven't read what they say to do yet)

---

... and here you could conditionally catch warnings or not depending on the MNE and/or At version if you want